network/adaptive_api.go

```diff
@@ -0,0 +1,45 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/ethersphere/swarm/log"
+)
+
+// CapabilitiesAPI abstracts RPC API access to capabilities controls
+// will in the future provide notifications of capability changes
+type CapabilitiesAPI struct {
+	*Capabilities
+}
+
+// NetCapabilitiesAPI creates new API abstraction around provided Capabilities object
+func NewCapabilitiesAPI(c *Capabilities) *CapabilitiesAPI {
+	return &CapabilitiesAPI{
+		Capabilities: c,
+	}
+}
+
+// RegisterCapability adds the given capability object to the Capabilities collection
+// If the Capability is already registered an error will be returned
+func (a *CapabilitiesAPI) RegisterCapability(cp *Capability) error {
+	log.Debug("Registering capability", "cp", cp)
+	return a.add(cp)
+}
+
+// IsRegisteredCapability returns true if a Capability with the given id is registered
+func (a *CapabilitiesAPI) IsRegisteredCapability(id CapabilityID) (bool, error) {
```

Comment:
maybe just `Has`?

Reply:
Remember this is in the namespace of `bzz`